modules/storages/lib/api/v3/storages/storage_representer.rb

```diff
@@ -81,6 +81,28 @@ def initialize(model, current_user:, embed_links: nil)
 
     property :name
 
+    property :applicationPassword,
+             skip_render: ->(*) { true },
+             getter: ->(*) {},
+             setter: ->(fragment:, represented:, **) {
+               if fragment.present?
+                 represented.automatically_managed = true
+                 represented.password = fragment
+               else
+                 represented.automatically_managed = false
+               end
+             }
+
+    property :hasApplicationPassword,
+             skip_parse: true,
+             skip_render: ->(represented:, **) { !represented.provider_type_nextcloud? },
+             getter: ->(represented:, **) {
+               break unless represented.provider_type_nextcloud?
```

Comment:
is this break needed, if you already define `skip_render` with the same condition?

Reply:
We need to guard against non-nextcloud models which do not define nextcloud provider_fields methods such as `#automatically_managed?` which is only defined in `Storages::NextcloudStorage`.

AFAICT from the unit tests `skip_render` is not exclusive of the `getter`. It seems both expressions are still evaluated; regardless of `skip_render` evaluating to `true`.

See failing tests after removing the break guard clause

```ruby
getter: ->(represented:, **) {
              #  break unless represented.provider_type_nextcloud?

               represented.automatically_managed?
             },
#
# docker compose exec backend-test bundle exec rspec modules/storages/spec/lib/api/v3/storages
  1) API::V3::Storages::StorageRepresenter rendering properties Automatically managed project folders with a non-Nextcloud storage does not include the property hasApplicationPassword
     Failure/Error: subject(:generated) { representer.to_json }

     NoMethodError:
       undefined method `automatically_managed?' for #<Storages::Storage id: nil, provider_type: "unknown", name: "Storage 18", host: "https://host18.example.com", creator_id: nil, created_at: nil, updated_at: nil, provider_fields: {}>
```